content/general/concepts/algorithm/terms/merge-sort/merge-sort.md

```diff
@@ -0,0 +1,143 @@
+
+1. **Initialize**: 
+   - Set indices `l`, `r`, and `k` to 0 to track the positions in `left`, `right`, and `data`.
+2. **Compare and Merge**:
+   - While both `left` and `right` have elements remaining:
+     - Compare the elements at `left[l]` and `right[r]`.
+     - Copy the smaller element into `data[k]` and increment the corresponding index (`l` or `r`).
+     - Increment `k` to move to the next position in `data`.
+3. **Copy Remaining Elements**:
+   - If any elements remain in `left`, copy them directly into the remaining positions in `data`.
+   - Similarly, if any elements remain in `right`, copy them into the remaining positions in `data`.
+
+### Implementation Pt.2
+
+The following example written in [Java](https://www.codecademy.com/learn/learn-java) shows an implementation of the second part of Merge Sort — sorting the subarrays and merging them back into the original array:
+
+```java
+public static void merge(int[] data, int[] left, int[] right){
+    //indexes of arrays left, right, and data
+    int l,r,k;
+    l = r = k = 0;
+    while(l < left.length && r < right.length && k < data.length){
+        //find min between left & right element and insert it
+        if(left[l] < right[r]){
+            data[k] = left[l];
+            l++;
+        }
+        else{
+            data[k] = right[r];
+            r++;
+        }
+        //increment index of data after insertion
+        k++;
+    }
+    //if elements still remain in arrays left or right, insert them into the data
+    while(l < left.length){
+        data[k] = left[l];
+        l++;
+        k++;
+    }
+    while(r < right.length){
+        data[k] = right[r];
+        r++;
+        k++;
+    }
+}
+
+```
+
+### Time Complexity Breakdown
+
+Overall Time Complexity: _O(n log n)_
+
+### Breakdown:
+
+1. **Divide**:
+   - The `mergeSort` function recursively divides the array into halves until each subarray has only one element.
+   - This splitting process has a time complexity of _O(log n)_ due to the repeated halving.
+
+2. **Conquer**:
+   - The base case (array of size 1) requires no sorting, so its complexity is _O(1)._
+
+3. **Combine**:
+   - The `merge` function merges two sorted subarrays into a single sorted array.
+   - It iterates through both subarrays once, comparing elements and copying them to the final array.
+   - This merging process takes _O(n)_ time, where n is the total number of merged elements.
+
+**Overall Time Complexity:**
+
+- The recursive calls to `mergeSort` create a [log n-level tree](https://www.geeksforgeeks.org/how-to-solve-time-complexity-recurrence-relations-using-recursion-tree-method/) calls. 
+  - The number of levels in this tree directly relates to how many times you can divide the array by 2 before reaching single-element subarrays. This is equivalent to the logarithm of the array's size (log n). For example, an array of 8 elements would have 3 levels (log2 (8) = 3). _Reference the picture below for a visual understanding._
```

Comment:
```suggestion
  - The number of levels in this tree directly relates to how many times the array by can be divided by 2 before reaching single-element subarrays. This is equivalent to the logarithm of the array's size (log n). For example, an array of 8 elements would have 3 levels (log2 (8) = 3). _Reference the picture below for a visual understanding._
```

Reply:
@SSwiniarski alittle typo, you would just need to remove `by` after "the array"